### ISSUE TITLE
documentation for blockingCallFromThread doesn't exactly say the given callable can return a Deferred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4517]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#4517|
|Type|enhancement|
|Created|2010-06-23 00:55:06Z|
|Branch|https://github.com/twisted/twisted/tree/blockingcallfromthread-docs-4517|

It makes an oblique reference to a callback chain, but it simply lists 'f' as 'any callable'.  It should go into more detail about Deferreds.

<details><summary>Searchable metadata</summary>

```
trac-id__4517 4517
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_blockingcallfromthread_docs_4517 branches/blockingcallfromthread-docs-4517
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1277254506000000 1277254506000000
changetime__1278941140000000 1278941140000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29666]) Branching to 'blockingcallfromthread-docs-4517'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

I updated the API docs.  The howto does already talk about a Deferred return value.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

1. Missing a news file.

2. The first sentence still doesn't mention the fact it supports non-Deferreds. Maybe say something like "block waiting for the result, and if the result is a Deferred, additionally wait for that to fire"

3. Phrasing on "@raise" is rather awkward, but I couldn't come up with anything sufficiently better.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Thanks.  I added a news file and reworked the intro sentence and the @raise section.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good - feel free to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [29775]) Merge blockingcallfromthread-docs-4517

Author: exarkun
Reviewer: itamar
Fixes: [#4517](https://github.com/twisted/twisted/issues/4517)

Improve the `blockingCallFromThread` documentation by explicitly
describing the Deferred interaction.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
